File: book_catalog/src/main/java/com/example/book_catalog/service/BookService.java
package com.example.book_catalog.service;

import com.example.book_catalog.domain.Book;

import java.util.List;

public interface BookService {
    List<Book> getALl();

    Book getById(Long booksId);

    Long create(String bookName);

    void update(Long bookId, String bookName);

    void remove(Long bookId);
}

File: 8_book_catalog_with_mongoDb/src/main/kotlin/com/otus/homework/book_catalog_with_mongodb/service/BookService.kt
package com.otus.homework.book_catalog_with_mongodb.service

import com.otus.homework.book_catalog_with_mongodb.dto.BookDtoToCreate
import com.otus.homework.book_catalog_with_mongodb.dto.BookDtoToUpdate
import com.otus.homework.book_catalog_with_mongodb.model.Book

interface BookService {

    fun getAll(): List<Book>

    fun getById(id: String): Book

    fun add(dto: BookDtoToCreate): Book

    fun update(dto: BookDtoToUpdate): Book

    fun deleteById(id: String)
}